rag/advanced_rag/harness/tools/search.py

```
@@ -277,6 +280,288 @@ async def bm25_search(tools, query: str, kb_ids: list[str] | None = None, top_n:
+        top_n=5,
+        compile_kwd=compile_kwd,
+    )
+    if not seed_rows:
+        return []
+
+    seed_names: set[str] = set()
+    for r in seed_rows.values():
+        try:
+            payload = json.loads(r.get("content_with_weight") or "{}")
+        except Exception:
+            continue
```

Ruff (0.16.0)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)